diff --git a/mccraft_core/src/schema.rs b/mccraft_core/src/schema.rs
index 09a1c9f..c014f63 100644
--- a/mccraft_core/src/schema.rs
+++ b/mccraft_core/src/schema.rs
@@ -14,15 +17,18 @@ table! {
             id -> Int4,
             for_recipe -> Int4,
         }
     }
 
     table! {
+        use diesel::sql_types::*;
+        use crate::sql::ItemTypeMapping;
+
         mccraft.items (id) {
             id -> Int4,
-            ty -> Item_type,
+            ty -> ItemTypeMapping,
             human_name -> Text,
             minecraft_id -> Text,
         }
     }
 
     table! {
